pkg/cloudprovider/provider/vsphere/provider.go

```diff
@@ -773,3 +771,82 @@ func (p *provider) MachineMetricsLabels(machine *v1alpha1.Machine) (map[string]s
 func (p *provider) SetMetricsForMachines(machines v1alpha1.MachineList) error {
 	return nil
 }
+
+func get(ctx context.Context, machine *v1alpha1.Machine, datacenterFinder *find.Finder, config *Config, operatingSystem providerconfig.OperatingSystem) (*object.VirtualMachine, error) {
```

Comment:
The function name does not really match its content :confused: 

How about extracting the "cleanup" code here into a `DeleteBrokenInstances` function which takes `virtualMachineList` as argument ?

Something like:
- fetch all vm's
- cleanup broken instances
- filter based on the VM fields

I would really like to split the retrieval & cleanup code as much as possible to avoid misunderstandings when reading the code.

Reply:
> https://github.com/kubermatic/kubermatic/pull/3484

How would you call it?

> How about extracting the "cleanup" code here into a DeleteBrokenInstances function which takes virtualMachineList as argument

I would like to first change the `cloudprovider` interface so that the `Get` also gets the `MachineCreateDeleteData`, because then we call call `Cleanup` in here. Any objections to that?

Reply:
> How would you call it?

For the time being:
`getAndCleanup` ?

> I would like to first change the cloudprovider interface so that the Get also gets the MachineCreateDeleteData, because then we call call Cleanup in here. Any objections to that?

sgtm